cmd/kubectl-gadget/snapshot/process.go

```diff
@@ -79,96 +52,19 @@ func NewProcessParser(outputConfig *utils.OutputConfig, processFlags *ProcessFla
 		"pid":       {},
 	}
 
-	return &ProcessParser{
-		BaseSnapshotParser: BaseSnapshotParser{
-			availableCols: availableCols,
-		},
-		outputConfig: outputConfig,
-		processFlags: processFlags,
-	}
-}
-
-func (p *ProcessParser) GetColumnsHeader() string {
-	requestedCols := p.outputConfig.CustomColumns
-	if len(requestedCols) == 0 {
-		requestedCols = []string{"node", "namespace", "pod", "container", "comm", "pid"}
-		if p.processFlags.showThreads {
-			requestedCols = []string{"node", "namespace", "pod", "container", "comm", "tgid", "pid"}
+	customRun := func(callback func([]gadgetv1alpha1.Trace) error) error {
+		config := &utils.TraceConfig{
+			BaseTraceConfig: commonsnapshot.NewProcessBaseTraceConfig(),
+			CommonFlags:     commonFlags,
 		}
-	}
 
-	return p.BuildSnapshotColsHeader(requestedCols)
-}
-
-func (p *ProcessParser) TransformEvent(e *types.Event) string {
-	var sb strings.Builder
-
-	switch p.outputConfig.OutputMode {
-	case utils.OutputModeColumns:
-		if p.processFlags.showThreads {
-			sb.WriteString(fmt.Sprintf("%s\t%s\t%s\t%s\t%s\t%d\t%d",
-				e.Node, e.Namespace, e.Pod, e.Container,
-				e.Command, e.Tgid, e.Pid))
-		} else {
-			sb.WriteString(fmt.Sprintf("%s\t%s\t%s\t%s\t%s\t%d",
-				e.Node, e.Namespace, e.Pod, e.Container,
-				e.Command, e.Pid))
-		}
-	case utils.OutputModeCustomColumns:
-		for _, col := range p.outputConfig.CustomColumns {
-			switch col {
-			case "node":
-				sb.WriteString(fmt.Sprintf("%s", e.Node))
-			case "namespace":
-				sb.WriteString(fmt.Sprintf("%s", e.Namespace))
-			case "pod":
-				sb.WriteString(fmt.Sprintf("%s", e.Pod))
-			case "container":
-				sb.WriteString(fmt.Sprintf("%s", e.Container))
-			case "comm":
-				sb.WriteString(fmt.Sprintf("%s", e.Command))
-			case "tgid":
-				sb.WriteString(fmt.Sprintf("%d", e.Tgid))
-			case "pid":
-				sb.WriteString(fmt.Sprintf("%d", e.Pid))
-			default:
-				continue
-			}
-			sb.WriteRune('\t')
-		}
+		return utils.RunTraceAndPrintStatusOutput(config, callback)
 	}
 
-	return sb.String()
-}
+	// It is not currently useful to pass processFlags here but it keeps
+	// uniformity with other gadgets.
```

Comment:
I meant that `processFlags` could be directly defined inside `NewCommonProcessCmd` because it is not used in  [`newProcessCmd`](https://github.com/kinvolk/inspektor-gadget/blob/25fbe811b2f4f5811f10c2bb4094067f0fa5eb7e/cmd/kubectl-gadget/snapshot/process.go#L57) as socket-collector [does](https://github.com/kinvolk/inspektor-gadget/blob/25fbe811b2f4f5811f10c2bb4094067f0fa5eb7e/cmd/kubectl-gadget/snapshot/socket.go#L59).